src/main/drivers/pwm_output.c

```diff
@@ -384,7 +384,7 @@ void pwmWriteDshotCommand(uint8_t index, uint8_t motorCount, uint8_t command)
         case DSHOT_CMD_SAVE_SETTINGS:
         case DSHOT_CMD_SPIN_DIRECTION_NORMAL:
         case DSHOT_CMD_SPIN_DIRECTION_REVERSED:
-            repeats = 10;
+            repeats = 20;
```

Comment:
I am strictly against watering down the standards here. It 10 and not 20 is what the protocol says, and it seems to be working fine with KISS,  the official BLHeli_S, and BLHeli32, so no point in forcing all of these too adapt to inofficial BLHeli_S to avoid double readings.

If anything, the workaround should be to introduce a `blheli_s_inofficial` setting to allow the user to enable this behaviour if their ESCs need it.

Reply:
This still doesn't take unreported CRC errors into account

Reply:
By changing the standard to require 20 repeats we will not be decreasing the probability of CRC errors.

I think the KISS / BLHeli_S official / BLHeli32 ESC firmware makes some allowances for CRC errors by allowing a certain number (1 in the case of BLHeli32 I believe) of the 10 expected repeats to be frames with CRC errors.

Reply:
If it allows at least one then ok, but requiring all 10 is too strict, hence the increase at FC side. I actually think 12 would be sufficient

Reply:
I didn't suggest changing the standard in any way, just amortizing CRC errors on FC size.

Reply:
In my opinion 'amortising' CRC errors on FC side is wrong. What should the ESC do in cases where there is no CRC error? Receive the command, and then signal an error because there was an incomplete command (the 2 repetitions that are outside of the spec) received?

Reply:
This is official blheli branch that is having issues 16.63.

Reply:
Ok. I tested with that and I have never seen issues.

Has this been raised as an issue with Blheli?

Also, are you sure that it's not a Betaflight issue? All releases before 3.2.0-RC4 had an issue that DMA was not properly reset when sending Dshot commands repeatedly. Any problem reports of 'commands only word for first <x> ESCs' are most likely due to this problem, and not ESC problems.

Reply:
If exactly 10 vs at least 10 is a sticking point I can see if I can put a pull request to require at least 5 identical commands (instead of at least 10) on official blheli_s so it is a little more robust then currently.

Reply:
It was from the tip of the master from yesterday.  Could be blheli_s or betaflight issue or a crc error. I wasn't able to track it down yet.

Reply:
I would go for 6 instead (to prevent the case where all frames being decoded properly leads to the command recognised as being sent twice). But adding a fudge factor of requiring (x / 2) +1  to x frames being received properly on the client side sounds reasonable to me. (Looking at the very low rate of frame CRC errors we have measured with KISS ESCs, even the requirement for 10 repeats seems to be high in hindsight, but this is what was decided on when the standard was set, and now that we have at least 5 different implementations of the standard, I think making it work as it is is better than starting to change it.

Reply:
How exactly did the problem manifest? And that was with BLHeli_S 16.63?

Reply:
Not all the motors would reverse everytime and not all of them would reverse back was the big issue (quad going nuts on the ground).  I'm sure it could happen to other 10 sequence commands as well. 

I hadn't had much of chance to test 16.63 since I was gone on vacation, but I had been getting messages from people that had issues.  I thought your recent fix might have solved it, but it appears there are situations where it still happens.

There are 2 problems the 10 repeats and the CRC are trying to solve.
1) Stop a bad packet from getting through either failing the CRC or in the more rare chance fail multiple times.  We doing this well (well maybe overdoing like you said)

2) Since this is going to send to multiple motors we need to make sure they will all receive it.  If 3 of the 4 switch bad things happen.  This could be due to noise, DMA bug, blheli bug.  Right not this isn't robust at all (at least with blheli_s the only source I can look at).  I can't say what kiss is doing here and it isn't in the spec (although we could test to find out exactly).  They just say to send 10 times, but they may only need a couple times to receive the command.  Luckily nothing bad happens if we receive the command 10 or 20 or 100 times, so sending it a bunch won't hurt anything (besides use up time, but we are at idle anyways).  Or on the ESC side blheli could only need 5 repeats and it should be just fine if it gets 2 sets of 5.

Reply:
Totally agreed on 1. and 2.. But for this to work, we need the ESC to allow for some rate of failure, and not expect exactly the number of repeats that are sent.

Just increasing the number of repetitions on the firmware side, while the ESC still expects a fixed number of repeats will not work.

Imagine this: If we send 20 repeats, and the ESC gets 14, with 6 lost due to noise, it will expect 6 more to complete the second iteration command. This will cause 6 of the repeats for the next command to be discarded as faulty. If the loss to noise stays at 33%, another 6 repeats of the second command will be lost, leaving only 8, which means the second command will never be recognised by the ESC. If use the same channel with 33% loss rate, but only send 10 repeats, but the ESC accepts anything between 6 and 10 repeats as valid command, this will not happen.